echopype/calibrate/env_params_old.py

```diff
@@ -105,8 +105,10 @@ def _apply(self, echodata) -> Dict[str, xr.DataArray]:
         if self.data_kind == "mobile":
             if np.isnan(echodata["Platform"]["time1"]).all():
                 raise ValueError("cannot perform mobile interpolation without time1")
+            # only grab needed variables for the interpolation
+            platform_data = echodata["Platform"][["latitude", "longitude"]]
             # compute_range needs indexing by ping_time
-            interp_plat = echodata["Platform"].interp(
+            interp_plat = platform_data.interp(
```

Comment:
Something is going on under the hood with xarray when calling `.interp` with all of the variables after version `2022.6.0`, which undergone the new `indexing`. By limiting the `.interp` to the data variables that are actually used, this works properly.

### All var
<img width="764" alt="Screenshot 2023-04-17 at 4 50 37 PM" src="https://user-images.githubusercontent.com/17802172/232634935-e030ff37-dc87-43bc-8c66-bcf97a44d627.png">

### Filtered var

<img width="764" alt="Screenshot 2023-04-17 at 4 50 50 PM" src="https://user-images.githubusercontent.com/17802172/232634968-3b740d1b-f4a4-4682-a291-660c6b99f7c3.png">

Reply:
Geez, ok, thanks for catching this. There may be some other parts in echopype that will break because of this that are not caught by the current tests, but we'll fix them when we get to it.